### PR TITLE
docs: Document clustermesh datapath configuration for non-tunneled modes

### PR DESCRIPTION
**Pull request by jrajahalme**
Add a section for cluster addressing requirements for non-tunneled
datapath modes.

Signed-off-by: Jarno Rajahalme <jarno@isovalent.com>

### CODE REVIEW

**Comment by jrajahalme** [comment]
test-docs

File: Documentation/gettingstarted/clustermesh/clustermesh.rst [hunk 1]
```diff
@@ -26,6 +26,39 @@ Cluster Addressing Requirements
 * The network between clusters must allow the inter-cluster communication. The
   exact ports are documented in the :ref:`firewall_requirements` section.
 
+Cilium installation may not default to a tunneled datapath mode in a specific
+cloud environment. This can be overridden with:
+
+ * ConfigMap option ``tunnel=vxlan``
+ * Helm option ``--set tunnel=vxlan``
+ * ``cilium install`` option ``--config tunnel=vxlan``
```
Comment (qmonnet) [inline Documentation/gettingstarted/clustermesh/clustermesh.rst]: Nitpick: The indent on this list items marks them as a “quote” (they're put in `<blockquote>` HTML tags)
```suggestion
* ConfigMap option ``tunnel=vxlan``
* Helm option ``--set tunnel=vxlan``
* ``cilium install`` option ``--config tunnel=vxlan``
```

**Review by nbusseneau — COMMENTED** [review]
Content: LGTM.

Formatting: I found it a bit harder to put the pieces back together now that we have 2 separate sections with extremely similar content, but the second one marked as "additional" when they actually are complementary. I would suggest we refactor the requirements with a structure akin to that:

- General requirements (e.g. "All clusters must be configured with the same datapath mode")
- Considerations specific to tunneled datapath.
  - I would probably argue we drop the bit about the `vxlan` options: I think it is not necessary to tell the user how to put clusters in tunneled mode in the requirements, rather he should already know if he uses tunneled mode or not and thus if the requirements apply or not.
- Considerations specific to native-routing.

What do you think?

**Comment by jrajahalme** [comment]
> * I would probably argue we drop the bit about the `vxlan` options: I think it is not necessary to tell the user how to put clusters in tunneled mode in the requirements, rather he should already know if he uses tunneled mode or not and thus if the requirements apply or not.

Users using defaults (Helm or `cilium install`) may not know what the datapath mode is going to be. I had to look into the cilium-cli source code to check this out, as it varies by the cloud platform. At the minimum we'd need a link to somewhere else in the docks that explains how to select the datapath mode. Not sure where we have it?

**Comment by nbusseneau** [comment]
I would say here: https://docs.cilium.io/en/v1.10/concepts/networking/routing/

**Comment by nbusseneau** [comment]
I still think we should address this, re-opening.

**Comment by pchaigno** [comment]
@nbusseneau If this is something we want to address, we should open an issue for it, rather than keeping the PR open.

File: Documentation/gettingstarted/clustermesh/clustermesh.rst (hunk 1 above)
Comment (jrajahalme) [reply Documentation/gettingstarted/clustermesh/clustermesh.rst]: Done, thanks!

**Comment by jrajahalme** [comment]
@nbusseneau Restructured as suggested, with links to datapath mode docs that you pointed out. This also made it logical to remove the tunnel mode setting instructions.

File: Documentation/gettingstarted/clustermesh/clustermesh.rst [hunk 2]
```diff
@@ -26,6 +30,28 @@ Cluster Addressing Requirements
...
+
+* Cilium in each cluster must be configured with a native routing CIDR that
+  covers all the PodCIDR ranges across all connected clusters. Cluster CIDRs are
+  typically allocated from the ``10.0.0.0/8`` private address space. When this
+  is a case a native routing CIDR such as ``10.0.0.0/9`` should cover all
+  clusters:
+
+ * ConfigMap option ``native-routing-cidr=10.0.0.0/9``
+ * Helm option ``--set nativeRoutingCIDR=10.0.0.0/9``
+ * ``cilium install`` option ``--native-routing-cidr=10.0.0.0/9``
```
Comment (nbusseneau) [inline Documentation/gettingstarted/clustermesh/clustermesh.rst:44]: FYI, these options are deprecated and going away in 1.12, so we should probably replace with the new ones:

```suggestion
 * ConfigMap option ``ipv4-native-routing-cidr=10.0.0.0/9``
 * Helm option ``--set ipv4NativeRoutingCIDR=10.0.0.0/9``
 * ``cilium install`` option ``--ipv4-native-routing-cidr=10.0.0.0/9``
```

**Review by nbusseneau — COMMENTED** [review]
LGTM, just one nit.

**Review by sayboras — APPROVED** [review]
Apart from one point highlighted by Nicolas, LGTM :100:

**Review by qmonnet — APPROVED** [review]

File: Documentation/gettingstarted/clustermesh/clustermesh.rst (hunk 2 above)
Comment (nbusseneau) [reply Documentation/gettingstarted/clustermesh/clustermesh.rst:44]: Fixed by https://github.com/cilium/cilium/pull/20408

File: Documentation/gettingstarted/clustermesh/clustermesh.rst [hunk 3]
```diff
@@ -26,6 +30,28 @@ Cluster Addressing Requirements
 * The network between clusters must allow the inter-cluster communication. The
   exact ports are documented in the :ref:`firewall_requirements` section.
 
+Additional Requirements for Native-routed Datapath Modes
+--------------------------------------------------------
+
+* Cilium in each cluster must be configured with a native routing CIDR that
+  covers all the PodCIDR ranges across all connected clusters. Cluster CIDRs are
+  typically allocated from the ``10.0.0.0/8`` private address space. When this
+  is a case a native routing CIDR such as ``10.0.0.0/9`` should cover all
```
Comment (joestringer) [inline Documentation/gettingstarted/clustermesh/clustermesh.rst:39]: @jrajahalme was there a reason that this lists `10.0.0.0/9` rather than `10.0.0.0/8`?

**Pull request merged**